src/StoreKeeper/WooCommerce/B2C/Frontend/Handlers/Seo.php

```diff
@@ -2,105 +2,45 @@
 
 namespace StoreKeeper\WooCommerce\B2C\Frontend\Handlers;
 
-use StoreKeeper\WooCommerce\B2C\Options\FeaturedAttributeOptions;
-use StoreKeeper\WooCommerce\B2C\Tools\FeaturedAttributes;
+use StoreKeeper\WooCommerce\B2C\Frontend\Handlers\Seo\StorekeeperHandler;
+use StoreKeeper\WooCommerce\B2C\Frontend\Handlers\Seo\YoastHandler;
+use StoreKeeper\WooCommerce\B2C\Options\StoreKeeperOptions;
 
 class Seo
 {
-    public function addExtraSeoData($markdown, $product)
-    {
-        // one of
-        /* @var $product \WC_Product_Variable */
-        /* @var $product \WC_Product_Simple */
-        $this->applyBrand($markdown, $product);
-        $this->applyBarcode($markdown, $product);
-
-        foreach ($markdown['offers'] as &$offer) {
-            if (isset($offer['priceSpecification']['valueAddedTaxIncluded'])) {
-                $offer['priceSpecification']['valueAddedTaxIncluded'] = true; // we always set with tax included
-            }
-            // seems google does not handle https:// and woommerce does not want to change
-            // see https://support.google.com/webmasters/thread/3210194?hl=en
-            if (!empty($offer['availability'])) {
-                $offer['availability'] = str_replace(
-                    'https://schema.org/',
-                    'http://schema.org/',
-                    $offer['availability']
-                );
-            }
+    const YOAST_HANDLER = 'yoast';
+    const STOREKEEPER_HANDLER = 'storekeeper';
+    const NO_HANDLER = 'none';
 
-            // get minimum price instead of first variation
-            if ($product instanceof \WC_Product_Variable) {
-                $offer['price'] = $product->get_variation_price('min', false);
-            }
+    public function prepareSeo($markdown, $product)
+    {
+        $selectedHandler = StoreKeeperOptions::get(StoreKeeperOptions::SEO_HANDLER, self::STOREKEEPER_HANDLER);
+        $handler = null;
+
+        switch ($selectedHandler) {
+            case self::STOREKEEPER_HANDLER:
+                $handler = new StorekeeperHandler();
+                break;
+            case self::YOAST_HANDLER:
+                $handler = new YoastHandler();
+                break;
```

Comment:
Hmm there is nothing to handle when yoost is rendering it.

Reply:
Removed `YoastHandler` as it doesn't handle anything and moved it to `Helpers\Seo`.